libraries/nodes/src/FFTNode.cpp

```diff
@@ -747,7 +747,7 @@ namespace nodes
         {
             _fftSize = _input.Size();
         }
-        _output.SetSize(_fftSize / 2);
+        _output.SetSize(_fftSize / 2 + 1);
```

Comment:
this is not complete, we'll also need to fix the Compile method to set this last value in the _output buffer, otherwise it will always be returned as a zero.  I'm not sure what that fix would look like since we are using the Cooley–Tukey algorithm with radix 2, meaning it is a divide by 2 strategy, so I'm not sure how this would ever set the last value in the output buffer.

Reply:
I checked the value and It was not zero and it was matching with the output of fft function in numpy.

Reply:
We don't need to change anything in the compile method. 
If we apply nfft of 512 to any input. What Cooley-Turkey algorithm is doing is It calculates the vector of 512 of which the end (512/2 +1 : 512) values are same as that of initial (1 : 512/2) in reverse order and the value at the [0], [512/2] is different. I am attaching a screenshot of matlab code calculating the FFT only which might help to clear (not calculating magnitude of complex number). So thats why all the fft functions return [0], [1: 512/2], [512]

Just FYI: index in matlab starts from 1 and not from 0.
https://www.mathworks.com/help/matlab/ref/fft.html
![Screenshot (2)](https://user-images.githubusercontent.com/17671208/59011421-4e11dc80-8852-11e9-9c1a-a1343aeed8f6.png)